connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java

```diff
@@ -264,17 +262,29 @@ void updateMetadataIfNeeded() {
     }
   }
 
-  Field updatedField(Field field, String marker) {
+  Field updatedField(Field field, StructField sparkSchemaField) {
     Field.Builder newField = field.toBuilder();
-    String description = field.getDescription();
-    if (description == null) {
-      newField.setDescription(marker);
-    } else if (!description.endsWith(marker)) {
-      newField.setDescription(description + " " + marker);
+
+    if (doesFieldContainsDescription(sparkSchemaField)) {
+      newField.setDescription(sparkSchemaField.metadata().getString("description"));
+    } else {
+      String description = field.getDescription();
+      String marker = SupportedCustomDataType.of(sparkSchemaField.dataType()).get().getTypeMarker();
+
+      if (description == null) {
+        newField.setDescription(marker);
+      } else if (!description.endsWith(marker)) {
+        newField.setDescription(description + " " + marker);
+      }
     }
     return newField.build();
   }
 
+  private boolean doesFieldContainsDescription(StructField field) {
+    return (field.metadata().contains("description")
```

Comment:
Please use the built in `StructField.getComment()`

Reply:
Here I am reading from the key "description" of the Metadata, whereas StructField.getComment() reads from the key "comment" of the Metadata. 
So we intend to support both the keys? or just one of the keys?
 if we intend to support both the keys and if user supplies different values to both the keys what shall we do?

Reply:
It was my mistake to force the BigQuery name "description" to Spark, we should support the standard `getComment()`. For backwards compatibility we will support both, where the standard  `getComment()` takes precedence.

Reply:
Notice the comment field should also be populated when reading from BigQuery

Reply:
Done